pkg/descheduler/strategies/lownodeutilization.go

```diff
@@ -274,28 +258,24 @@ func evictPodsFromTargetNodes(
 	sortNodesByUsage(targetNodes)
 
 	// upper bound on total number of pods/cpu/memory to be moved
-	totalAvailableUsage := map[v1.ResourceName]*resource.Quantity{
-		v1.ResourcePods:   {},
-		v1.ResourceCPU:    {},
-		v1.ResourceMemory: {},
-	}
+	totalAvailableUsage := map[v1.ResourceName]*resource.Quantity{}
 
 	var taintsOfLowNodes = make(map[string][]v1.Taint, len(lowNodes))
 	for _, node := range lowNodes {
 		taintsOfLowNodes[node.node.Name] = node.node.Spec.Taints
 
-		for name := range totalAvailableUsage {
+		for name := range node.highResourceThreshold {
+			if _, ok := totalAvailableUsage[name]; !ok {
+				totalAvailableUsage[name] = resource.NewMilliQuantity(0, resource.DecimalSI)
+			}
 			totalAvailableUsage[name].Add(*node.highResourceThreshold[name])
 			totalAvailableUsage[name].Sub(*node.usage[name])
 		}
 	}
 
-	klog.V(1).InfoS(
-		"Total capacity to be moved",
-		"CPU", totalAvailableUsage[v1.ResourceCPU].MilliValue(),
-		"Mem", totalAvailableUsage[v1.ResourceMemory].Value(),
-		"Pods", totalAvailableUsage[v1.ResourcePods].Value(),
-	)
+	for name := range totalAvailableUsage {
+		klog.V(1).InfoS("Total capacity to be moved (in milli)", name, totalAvailableUsage[name].MilliValue())
```

Comment:
This is going to add a lot of `Total capacity to be moved` to the logs, could you maybe log that part once and then log each of the resources?

Reply:
You can keep `Total capacity to be moved` as it is and log the extended resources alongside if they are present.